PlayerInfo/PlayerInfo.cpp

```diff
@@ -85,19 +89,58 @@ namespace Plugin {
         return message;
     }
 
+    void PlayerInfo::Deactivated(RPC::IRemoteConnection* connection)
+    {
+        if (connection->Id() == _connectionId) {
+            ASSERT(_service != nullptr);
+            // This can potentially be called on a socket thread, so the deactivation (wich in turn kills this object) must be done
+            // on a seperate thread. Also make sure this call-stack can be unwound before we are totally destructed.
+            Core::IWorkerPool::Instance().Submit(PluginHost::IShell::Job::Create(_service, PluginHost::IShell::DEACTIVATED, PluginHost::IShell::FAILURE));
+        }
+    }
+
     /* virtual */ void PlayerInfo::Deinitialize(PluginHost::IShell* service)
     {
         ASSERT(_player != nullptr);
-        if (_player != nullptr) {
-            Exchange::JPlayerProperties::Unregister(*this);
-            _player->Release();
-        }
+        ASSERT(_audioCodecs != nullptr);
+        ASSERT(_videoCodecs != nullptr);
+        ASSERT(service == _service);
+
+        _service->Unregister(&_rcnotification);
+
         if (_dolbyOut != nullptr)
         {
              _notification.Deinitialize();
             Exchange::Dolby::JOutput::Unregister(*this);
+
+            _dolbyOut->Release();
+            _dolbyOut = nullptr;
+
         }
+
+        _audioCodecs->Release();
+        _audioCodecs = nullptr;
+        _videoCodecs->Release();
+        _videoCodecs = nullptr;
+
+        Exchange::JPlayerProperties::Unregister(*this);
+        auto const result = _player->Release();
+
+        if (result != Core::ERROR_DESTRUCTION_SUCCEEDED) {
+            RPC::IRemoteConnection* connection(_service->RemoteConnection(_connectionId));
+
+            // The process can disappear in the meantime...
+            if (connection != nullptr) {
+
+                // But if it did not dissapear in the meantime, forcefully terminate it. Shoot to kill :-)
+                connection->Terminate();
+                connection->Release();
+            }
```

Comment:
line 137 - connection = nullptr; ??
my understanding is that after terminate and release it is no longer for use and rest of the code may rely on connection ptr

Reply:
The connection variable is declared in the if scope.